File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        sol = {}
        for i in range(len(nums)):
            k = target - nums[i]
            if k in sol:
                return [sol[k], i]
            sol[nums[i]] = i

File: 0001-two-sum/0001-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    
    let sol ={};
    for(let i = 0; i<nums.length; i++){
        
        let k = target - nums[i];
        if (k in sol){
            return [sol[k], i];
        }
        sol[nums[i]] = i;
    }
    return [];
}